modules/airavata-helix/task-core/src/main/java/org/apache/airavata/helix/core/participant/HelixParticipant.java

```diff
@@ -60,30 +58,40 @@
     private String clusterName;
     private String participantName;
     private ZKHelixManager zkHelixManager;
+
     private String taskTypeName;
     private PropertyResolver propertyResolver;
-    private Class<T> taskClass;
+
+    private List<Class<? extends T>> taskClasses;
     private final List<AbstractTask> runningTasks = Collections.synchronizedList(new ArrayList<AbstractTask>());
 
-    public HelixParticipant(Class<T> taskClass, String taskTypeName) throws ApplicationSettingsException {
+    public HelixParticipant(List<Class<? extends T>> taskClasses, String taskTypeName) throws ApplicationSettingsException {
```

Comment:
This is not relevant to this PR. However please enable the participant manually when it is registering in to the cluster. Participant become disabled when they are gracefully turned off and do not go into enabled state when they come back to the cluster

Reply:
I have updated the PR with a modification to the HelixParticipant. @DImuthuUpe can you verify whether this addresses the actual requirement you have mentioned?

Reply:
@DImuthuUpe I tested the implementation in my local environment and it is working as expected.